internal/platform/gin/drain.go

```diff
@@ -22,22 +22,29 @@ import (
 	"github.com/gin-gonic/gin"
 	"github.com/goph/emperror"
 	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
 	"github.com/spf13/viper"
 )
 
 var basePath = viper.GetString("pipeline.basepath")
 
 // DrainModeMiddleware prevents write operations from succeeding.
 type DrainModeMiddleware struct {
-	enabled bool
-	mu      sync.RWMutex
+	enabled   bool
+	mu        sync.RWMutex
+	drainMode prometheus.Gauge
```

Comment:
Call it maybe drainModeMetric or drainModeGauge or something that tells us what this is.